packages/fxa-settings/src/pages/Index/container.tsx

```diff
@@ -174,16 +151,24 @@ export const IndexContainer = ({
           }
         }
 
-        handleNavigation(exists, hasLinkedAccount, hasPassword, email, true);
+        handleSuccess(exists, hasLinkedAccount, hasPassword, email, isSubmit);
         return { error: null };
       } catch (error) {
         GleanMetrics.emailFirst.submitFail();
         return getHandledError(error);
       }
     },
-    [authClient, handleNavigation, isWebChannelIntegration]
+    [authClient, handleSuccess, isWebChannelIntegration]
   );
 
+  useEffect(() => {
+    if (!hasEmailSuggestion) {
+      return;
+    }
+
+    signUpOrSignInHandler(suggestedEmail, false);
```

Comment:
I noticed that we weren't checking for email validity before redirecting to signup, so modified to use the signUpOrSignInHandler that includes the checks then navigates.